utilities/avro-schema-generator/src/main/java/com/flipkart/aesop/avro/schemagenerator/data/MysqlToAvroMapper.java

```diff
@@ -94,16 +98,26 @@
 	TIME("long"),
 
 	/** The year. */
-	YEAR("long");
+	YEAR("long"),
+
+	/** For HashMap.
+	 * Though this Mysql Datatype does exist, this required for
+	 * passing record changes in form of Avro Map datatype.
+	 */
+	MAP(new HashMap<String, Object>(){{
```

Comment:
Use Guava's Immutable Map and Immutable list.

I need to understand this more. Let's catch up.